tools/cldr-code/src/main/java/org/unicode/cldr/util/UnitConverter.java

```diff
@@ -241,23 +241,113 @@ public UnitConverter cloneAsThawed() {
+            if (special != null) {
+                if (special.equals("beaufort")) {
+                    return (inverse) ? beaufortToBase(source) : baseToBeaufort(source);
+                }
+                return source;
+            }
             return source.subtract(offset).divide(factor);
         }
 
+        private static final BigInteger BEAUFORT_MAX = BigInteger.valueOf(17);
+        private static final Rational[] beaufortToMetersPerSec = {
+            // midRange m/s values to use for different Bft values
+            // mostly from m/s = 0.836*(B^1.5) but adjust value for Bft 0
+            Rational.of("0.1"), // 0 Bft
+            Rational.of("0.8"), // 1
+            Rational.of("2.4"), // 2
+            Rational.of("4.3"), // 3
+            Rational.of("6.7"), // 4
+            Rational.of("9.3"), // 5
+            Rational.of("12.3"), // 6
+            Rational.of("15.5"), // 7
+            Rational.of("18.9"), // 8
+            Rational.of("22.6"), // 9
+            Rational.of("26.4"), // 10
+            Rational.of("30.5"), // 11
+            Rational.of("34.8"), // 12
+            Rational.of("39.2"), // 13
+            Rational.of("43.8"), // 14
+            Rational.of("48.6"), // 15
+            Rational.of("53.5"), // 16
+            Rational.of("58.6") // 17
+        };
+        private static final Rational[] minMetersPerSecForBeaufort = {
+            // minimum m/s values for each Bft values
+            // from table in Wikipedia
+            Rational.of("0.0"), // 0 Bft
+            Rational.of("0.3"), // 1
+            Rational.of("1.6"), // 2
+            Rational.of("3.4"), // 3
+            Rational.of("5.5"), // 4
+            Rational.of("8.0"), // 5
+            Rational.of("10.8"), // 6
+            Rational.of("13.9"), // 7
+            Rational.of("17.2"), // 8
+            Rational.of("20.8"), // 9
+            Rational.of("24.5"), // 10
+            Rational.of("28.5"), // 11
+            Rational.of("32.7"), // 12
+            Rational.of("36.9"), // 13
+            Rational.of("41.4"), // 14
+            Rational.of("46.1"), // 15
+            Rational.of("51.1"), // 16
+            Rational.of("55.8") // 17
+        };
+
+        private Rational beaufortToBase(Rational beaufort) {
+            BigInteger beaufortRound = beaufort.abs().add(Rational.of(1, 2)).floor();
+            if (beaufortRound.compareTo(BEAUFORT_MAX) > 0) {
+                beaufortRound = BEAUFORT_MAX;
+            }
+            int beaufortIndex = beaufortRound.intValue();
+            return beaufortToMetersPerSec[beaufortIndex];
+        }
+
+        private Rational baseToBeaufort(Rational base) {
+            base = base.abs();
+            int beaufortIndex = BEAUFORT_MAX.intValue();
+            while (base.compareTo(minMetersPerSecForBeaufort[beaufortIndex]) < 0) {
+                beaufortIndex--;
+            }
+            return Rational.of(beaufortIndex);
```

Comment:
This should work fine, but just to mention that Java has https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#binarySearch-java.lang.Object:A-java.lang.Object-

Reply:
Yes, as I mentioned to Robert,  the binary search would require two comparisons per iteration (both ends of range) instead of one in the linear search; since the size is relatively small,I thought it might be a wash.

Reply:
I was thinking more of code simplicity. For the CLDR reference implementation, accuracy and simplicity are much more important than shaving a few cycles off.

Reply:
Done, I did end up switching to use Arrays.binarySearch